catalog/content/core/content-core-catalogerplugin/src/test/java/ddf/content/plugin/cataloger/TestCatalogContentPlugin.java

```diff
@@ -0,0 +1,505 @@
+        when(mockSubject.getPrincipals()).thenReturn(null);
+
+        mockContentCreateResponse.getProperties().put(SecurityConstants.SECURITY_SUBJECT, new HashMap<>());
+        CreateResponse response = catalogContentPlugin.process(mockContentCreateResponse);
+        assertThat(response, notNullValue());
+
+        assertThat(response.getCreatedMetadata(), is(mockContentCreateResponse.getCreatedMetadata()));
+        assertThat(response.getCreatedContentItem(), is(mockContentCreateResponse.getCreatedContentItem()));
+        assertThat(response.getCreatedMetadataMimeType(), is(mockContentCreateResponse.getCreatedMetadataMimeType()));
+    }
+
+    @Test
+    public void testAddAttributesFromStoragePlugin() throws Exception {
+        Map<String, Serializable> propMap = new HashMap<>();
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put("Test", "Abc123");
+        propMap.put(ContentPlugin.STORAGE_PLUGIN_METACARD_ATTRIBUTES, properties);
+        when(mockContentCreateResponse.getProperties()).thenReturn(propMap);
+
+        CreateResponse response = catalogContentPlugin.process(mockContentCreateResponse);
+        assertThat(response, notNullValue());
+        assertThat(response.getCreatedMetadata(), is(mockContentCreateResponse.getCreatedMetadata()));
+        assertThat(response.getCreatedContentItem(), is(mockContentCreateResponse.getCreatedContentItem()));
+        assertThat(response.getCreatedMetadataMimeType(), is(mockContentCreateResponse.getCreatedMetadataMimeType()));
+    }
+
+    @Test
+    public void testMetacardNoTitle() throws Exception {
+        Metacard noTitleCard = new MetacardImpl();
+        ((MetacardImpl)noTitleCard).setId(UUID.randomUUID().toString());
+
+        List<Metacard> noTitleList = new ArrayList<>();
+        noTitleList.add(noTitleCard);
+
+        when(mockContentItem.getUri()).thenReturn(null);
+        when(mockCatalogCreateResponse.getCreatedMetacards()).thenReturn(noTitleList);
+        when(mockInputTransformer.transform(any(InputStream.class))).thenReturn(noTitleCard);
+        when(mockInputTransformer.transform(any(InputStream.class), any(String.class))).thenReturn(noTitleCard);
+
+        CreateResponse response = catalogContentPlugin.process(mockContentCreateResponse);
+        assertThat(response, notNullValue());
+        assertThat(response.getCreatedMetadata(), is(mockContentCreateResponse.getCreatedMetadata()));
+        assertThat(response.getCreatedContentItem(), is(mockContentCreateResponse.getCreatedContentItem()));
+        assertThat(response.getCreatedMetadataMimeType(), is(mockContentCreateResponse.getCreatedMetadataMimeType()));
+    }
+
+    @Test
+    public void testDynamicMetacard() throws Exception {
+        DynamicMetacard dynamicMetacard = metacardFactory.newInstance();
+        dynamicMetacard.addAttribute(Metacard.TITLE, "Dynamic Metacard");
+        dynamicMetacard.addAttribute("Attrib1", "Test Value");
```

Comment:
Since "Test Value", "Attrib1", and "Test Exception" are used in multiple spots, it may be better to have them as global static final constants